docs/data-sources/repository_credential.md

```diff
@@ -0,0 +1,153 @@
+---
+Title: "Repository Credential Data Source"
+Description: |-
+    Creating the Repository Credential data source.
+---
+
+# Repository Credential
+
+The `tanzu-mission-control_repository_credential` data source allows you to get repository credential to a particular scope through Tanzu Mission Control.
+
+[repository-credential]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-657661A2-B26E-412A-9A46-7467A44A075A.html
+
+## repository-credential Scope
+
+In the Tanzu Mission Control resource hierarchy, there are two levels at which you can specify repository credential resources:
+- **object groups** - `cluster_group` block under `scope` sub-resource
+- **Kubernetes objects** - `cluster` block under `scope` sub-resource
+
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script for the provider to function.
+Only one scope per resource is allowed.
+
+
+## Cluster group scoped Repository Credential
+
+### Example Usage
+
+```terraform
+# Read Tanzu Mission Control source secret : fetch cluster group source secret details
+data "tanzu-mission-control_repository_credential" "read_source_secret" {
```

Comment:
Please differentiate the resource name based on the scope here and at other places needed